vcfflatten: fix segfault when no 'AF' field is present (#47).

mphschmitt opened the pull request:
vcfflatten crashed when flattening a VCF file with more than 2 alleles for a locus if no 'AF' field was present.

here are the offending lines:
```
#vcfflaten.cpp l.122

            vector<double>::iterator f = freqs.begin();   
            for (vector<string>::iterator a = var.alt.begin(); a != var.alt.end(); ++a, ++f) {   
                alleleFrequencies.insert(pair<double, string>(*f, *a));    
            }     
```
With no 'AF' field, we ended up with an empty freqs vector. Some pointer arithmetic then incremented the pointer f, which was finally dereferenced to an illegal value.

To solve this, I propose to terminate the program when a line with more than 2 alleles for a locus is encountered if it has no 'AF' field.
However, we can add the --best-effort option. If it is passed as an argument to the program alongside a filename, then we output the offending line without modifying it.

mphschmitt committed:
vcfflatten: fix segfault when no 'AF' field is present (#47).

vcfflatten crashed when 'flattening' a VCF file with more
than 2 alleles for a locus if no 'AF' field was present.
Add a new option to handle this behavior:
  - If the --ignore-errors option is passed to the program,
    do not try to flatten the offending line, and parse the
    next one.
  - if the --ignore-errors option is not passed, then the
    default behavior is to print an error message and
    exit.

pjotrp commented:
lgtm! There is a typo `--best-effor` in the help text. 

I like the idea of an `--ignore-errors` switch which overrides errors by either skipping or displaying unchanged info. You could use that instead. @ekg what do you think?

I'll merge either way.

mphschmitt committed:
vcfflatten: update man page.

Add the --help and --ignore-errors options to the man page.

mphschmitt commented:
I renamed the option to --ignore-errors, and reworked the parsing of the options a little bit so that the --ignore-errors option can still be used when reading from the stdin.
I also updated the manual accordingly.

pjotrp commented:
Thanks!
